x/nodes/keeper/valStateChanges.go

```diff
@@ -228,10 +238,18 @@ func (k Keeper) ValidateEditStake(ctx sdk.Ctx, currentValidator, newValidtor typ
 		}
 	}
 	if k.Cdc.IsAfterNonCustodialUpgrade(ctx.BlockHeight()) {
-		// ensure output address doesn't change
-		if currentValidator.OutputAddress != nil {
-			if !newValidtor.OutputAddress.Equals(currentValidator.OutputAddress) {
-				fmt.Println(currentValidator.String())
+		if k.Cdc.IsAfterOutputAddressChangeUpgrade(ctx.BlockHeight()) {
+			// ensure only the current output address owner can change the output address
+			if currentValidator.OutputAddress != nil &&
+				!signer.Equals(currentValidator.OutputAddress) &&
+				!newValidtor.OutputAddress.Equals(currentValidator.OutputAddress) {
+				return types.ErrDisallowedOutputAddressChange(k.Codespace())
+			}
+		} else {
+			//check the "new" validator's signature validity
+			//will recheck if validator exists
```

Comment:
Why not keep this as `// ensure output address doesn't change` like before? It's pretty clear.
```suggestion
			// check the "new" validator's signature validity
			// will re-check if validator exists
```